### PR TITLE
Update retrofit from 2.9.0 to 2.11.0

### PR DESCRIPTION
**Pull request by SimonMarquis**
- Replace Jake Wharton's Kotlinx Serialization converter with new retrofit's `converter-kotlinx-serialization`.
- Remove custom Proguard rules that are now part of the dependency: https://github.com/square/retrofit/tree/trunk/retrofit-response-type-keeper#readme

Changelog: https://github.com/square/retrofit/blob/trunk/CHANGELOG.md

This is an updated version of the initial update PR that remained untouched for too long:
- #1344

### CODE REVIEW

**Review by dturner — APPROVED** [review]

File: app/dependencies/prodReleaseRuntimeClasspath.txt [hunk 1]
```diff
@@ -190,12 +190,12 @@ com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
...
 com.google.protobuf:protobuf-javalite:4.26.1
 com.google.protobuf:protobuf-kotlin-lite:4.26.1
-com.jakewharton.retrofit:retrofit2-kotlinx-serialization-converter:1.0.0
+com.squareup.retrofit2:converter-kotlinx-serialization:2.11.0
 com.squareup.okhttp3:logging-interceptor:4.12.0
 com.squareup.okhttp3:okhttp:4.12.0
 com.squareup.okio:okio-jvm:3.8.0
 com.squareup.okio:okio:3.8.0
-com.squareup.retrofit2:retrofit:2.9.0
+com.squareup.retrofit2:retrofit:2.10.0
```
Comment (dturner) [inline app/dependencies/prodReleaseRuntimeClasspath.txt:198]: Should this be `2.11.0`?

**Comment by dturner** [comment]
Not sure why CI didn't run on this PR. Please could you update the badging to see if that kicks off the `Build` CI workflow?

File: app/dependencies/prodReleaseRuntimeClasspath.txt (hunk 1 above)
Comment (SimonMarquis) [inline app/dependencies/prodReleaseRuntimeClasspath.txt:198]:
```suggestion
com.squareup.retrofit2:retrofit:2.11.0
```

**Comment by SimonMarquis** [comment]
@dturner not sure why, but the summray still says: 

> This workflow requires approval from a maintainer. [Learn more about approving workflows.](https://docs.github.com/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks)
1 successful check

**Comment by SimonMarquis** [comment]
I'll close/re-open the PR.

**Review by dturner — APPROVED** [review]

**Comment by dturner** [comment]
I had to create a separate PR to resolve the merge conflicts. https://github.com/android/nowinandroid/pull/1623

**Pull request closed without merging**